src/Utility/OrderStatusUtility.php

```diff
@@ -52,4 +56,25 @@ public static function transformPaymentStatusToPaid($status, $comparedStatus)
 
         return $status;
     }
-}
\ No newline at end of file
+
+    /**
+     * @param MolliePaymentAlias|MollieOrderAlias $transaction
+     */
+    public static function transformPaymentStatusToRefunded($transaction)
+    {
+        if ($transaction->amountRefunded === null) {
+            return $transaction->status;
+        }
+
+        $amountRefunded = (float)$transaction->amountRefunded->value;
+        $amountPayed = (float)$transaction->amountCaptured->value;
+
+        if ($amountRefunded < $amountPayed) {
+            return Config::PARTIAL_REFUND_CODE;
+        } elseif ($amountRefunded === $amountPayed) {
+            return RefundStatus::STATUS_REFUNDED;
+        } else {
+            return $transaction->status;
+        }
```

Comment:
number comparison in PHP is tricky thing, there is in PrestaShop 1.7  `new Number()` class which does the heavy lifting of comparing floating point numbers. Can you check if we have that on PS 1.6 - if so, lets discuss further how we can approach this

Reply:
Checked 1.6 prestashop doesn't have "Number" class.